src/components/include/utils/logger.h

```diff
@@ -140,36 +140,66 @@ namespace logger {
 #define CREATE_LOGGERPTR_LOCAL(logger_var, logger_name) \
     std::string logger_var(logger_name);
 
-#define LOG4CXX_IS_TRACE_ENABLED(logger)
+#define LOG_WITH_LEVEL(loggerPtr, logLevel, logEvent, line) \
+do { \
+     std::stringstream accumulator; \
+     accumulator << __FILE__ << ":" << line << " " << __FUNCTION__ << ": " << logEvent; \
+     logger::push_log(loggerPtr, logLevel, logger::time_now(), accumulator.str()); \
+} while (false)
 
-#define LOG_WITH_LEVEL(loggerPtr, logLevel, logEvent) \
+#define LOG_WITH_LEVEL_EXT(loggerPtr, logLevel, logEvent, line, file, function) \
 do { \
      std::stringstream accumulator; \
-     accumulator << logEvent; \
+     accumulator << file << ":" << line << " " << function << ": " << logEvent; \
```

Comment:
In windows & qt versions we have no log4cxx. And this header is shared among all platforms. And this macro is used on win platform.